src/jobs/models/models.ts

```diff
@@ -1,11 +1,15 @@
 import { Prisma } from '@prismaClient';
 import type { components, operations } from '@src/openapi';
+import { StagePrismaObject } from '@src/stages/models/models';
 
 type JobModel = components['schemas']['jobResponse'];
 type JobCreateModel = components['schemas']['createJobPayload'];
 type JobCreateResponse = components['schemas']['createJobResponse'];
 type JobGetParams = components['parameters'];
 type JobFindCriteriaArg = operations['findJobs']['parameters']['query'];
-type JobPrismaObject = Prisma.JobGetPayload<{ include: { stage: boolean } }>;
 
-export type { JobModel, JobCreateModel, JobCreateResponse, JobGetParams, JobFindCriteriaArg, JobPrismaObject };
+type JobPrismaObject = Prisma.JobGetPayload<{ include: { stage: true } }> | Prisma.JobGetPayload<{ include: { stage: false } }>;
+
+type JobPrismaObjectWithStages = JobPrismaObject & { stage?: StagePrismaObject[] };
```

Comment:
maybe using generics would be nicer? give it a try for a few min
jobPrismaObject<includeStages extends boolean>

Reply:
done